### DISCUSSION TITLE
Sync-Waves using ApplicationSets

### DISCUSSION THREAD

**Discussion by RumRogerz** [argoproj/argo-cd#7315] [General]
Probably a silly question, but I just need some clarity.

I have 3 relatively heavily templated ApplicationSets that are responsible for delivering 3 Apps across different clusters, regions, space-time, etc.  

for simplicity:
```
app1 - db
app2 - another db
app3 - main app
```

`app3` is heavily reliant on `app1` and `app2` to be already up and synced.  It doesn't matter if `app2` or `app1` come up first - we just need `app3` to be the last one to start syncing.  

I have looked into sync-waves and the only examples I can see are with an App of Apps solution.  What I love about ApplicationSets is how easy it is to get things to scale without having to create a new Application manifest for it.  From what I've been seeing, I can't get any concrete answers as to whether this is possible with ApplicationSets.  App of Apps would be murder - the sheer amount of manifests that would need to be created would be an absolute nightmare. 

So, I guess ultimately, can I use sync-waves with ApplicationSets or am I doomed?

**Comment by ishitasequeira** [comment]
@RumRogerz right now sync-waves is not supported with ApplicationSet.

**Reply by RumRogerz** [reply]
I figured as much.  Thank you for clarifying that.

**Comment by initcron** [comment]
you could do this with Kargo now https://github.com/akuity/kargo

**Comment by rumstead** [comment]
Progressive syncs could help with your use case. https://argo-cd.readthedocs.io/en/latest/operator-manual/applicationset/Progressive-Syncs/
